core/src/inputs/advice/merkle_path_set.rs

```diff
@@ -0,0 +1,246 @@
+use super::{
+    hasher::{self, Digest},
+    AdviceSetError, Word,
+};
+use math::fields::f64::BaseElement;
+use std::collections::HashMap;
```

Comment:
We should probably use a `BTreeMap` from the `utils` crate here. It provides deterministic results.